src/Microsoft.Framework.Configuration.Binder/ConfigurationBinder.cs

```diff
@@ -12,28 +12,14 @@ namespace Microsoft.Framework.Configuration
 {
     public static class ConfigurationBinder
     {
-        public static TModel Bind<TModel>(IConfiguration configuration) where TModel : new()
+        public static void Bind(this IConfiguration configuration, object model)
```

Comment:
A couple of thoughts on this:
1. I completely agree we just need to reflect on the instance for the method that takes an instance. 
2. For the public surface experience I still like the API that binds to return the object, e.g.: 
 ``` C#
 var boundCat = config.Bind(new Cat);
 ```
 I understand that this is just convenience but I still like it and the generics are useful for that. 
3. Given that we do activate instances for non-scalar properties, it seems artificially restrictive not to have an overload that can automatically create the root object (i.e. the overload that you are removing here). I understand that I have changed my mind more than once on this. Sorry about that :smile: